### PR TITLE
[ci] [python-package] [R-package] adapt to scikit-learn check_sample_weight_equivalence changes, stop testing against R 3.6 on Linux

### PR DESCRIPTION
**Pull request by jameslamb**
More changes to estimator checks have been made in `scikit-learn`, and those broke CI here.

CI is failing like this when using the latest `scikit-learn` nightlies:

```text
=========================== short test summary info ============================
FAILED tests/python_package_test/test_sklearn.py::test_sklearn_integration[LGBMClassifier()-check_sample_weight_equivalence_on_dense_data]
FAILED tests/python_package_test/test_sklearn.py::test_sklearn_integration[LGBMClassifier()-check_sample_weight_equivalence_on_sparse_data]
===== 2 failed, 767 passed, 40 skipped, 2 xfailed, 584 warnings in 36.86s ======
```

([build link](https://github.com/microsoft/LightGBM/actions/runs/12076718415/job/33678491487#step:4:578))

Looks like https://github.com/scikit-learn/scikit-learn/pull/30137 removed `check_sample_weight_equivalence`, in favor of 2 new functions names `check_sample_weight_equivalence_on_{dense,sparse}_data`. This updates `lightgbm`'s xfail_checks tag to include all 3 names, so those tests are correctly xfail'd for the full range of `scikit-learn` versions `lightgbm` supports.

**update: this also removes the R 3.6 Linux CI job to fix another blocking CI issue, see https://github.com/microsoft/LightGBM/pull/6733#discussion_r1863011937**

### CODE REVIEW

**Comment by jameslamb** [comment]
Noticing that the R 3.6 CI job has been failing for a couple days like this:

> Post job cleanup.
/usr/bin/docker exec  0f363447ec94ed26e9ae21b628f3850fe9f1e15a4c8a341b96fcdc540018e374 sh -c "cat /etc/*release | grep ^ID"
/__e/node20/bin/node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /__e/node20/bin/node)

([build link](https://github.com/microsoft/LightGBM/actions/runs/12079388100/job/33685216855?pr=6733#step:21:3))

That's happening here:

https://github.com/microsoft/LightGBM/blob/83c0ff3de1925b0e2d4831a9ccb6ffc196aa795b/.github/workflows/r_package.yml#L176-L180

Will have to fix that in this PR too. I'll try that.

File: .github/workflows/r_package.yml [hunk 1]
```diff
@@ -48,12 +44,6 @@ jobs:
           ################
           # CMake builds #
           ################
-          - os: ubuntu-latest
-            task: r-package
-            compiler: gcc
-            r_version: 3.6
-            build_type: cmake
-            container: 'ubuntu:18.04'
```
Comment (jameslamb) [inline .github/workflows/r_package.yml:56]: Trying to fix https://github.com/microsoft/LightGBM/pull/6733#issuecomment-2507125349, it seems that the workaround from https://github.com/actions/checkout/issues/1590#issuecomment-2207052044 is no longer working.

See https://github.com/actions/checkout/issues/1590#issuecomment-2507159983.

Between this and all the work that was required with package-pinning in #6718, I think we should just fully stop testing R 3.6 on Linux in CI here. As we did for macOS in #5859.

Note that we'll still have test coverage on Windows:

https://github.com/microsoft/LightGBM/blob/83c0ff3de1925b0e2d4831a9ccb6ffc196aa795b/.github/workflows/r_package.yml#L81-L87

https://github.com/microsoft/LightGBM/blob/83c0ff3de1925b0e2d4831a9ccb6ffc196aa795b/.github/workflows/r_package.yml#L95-L102

Dropping this testing gets us the following:

* reduces CI configuration complexity (see all the code removed in this PR)
* allows upgrading to `actions/checkout@v4` (no more skipping PRs like #6712)
* reduced ongoing maintenance burden, via avoiding things like these:
  - #6718
  - #6298
  - #5903

**Review by StrikerRUS — APPROVED** [review]
LGTM! Thank you very much for quick fix! Nice to see this simplification diff for R.

**Comment by jameslamb** [comment]
Thanks very much for the quick review! Merging this to unblock CI.

**Pull request merged**